app/models/ticket_purchase.rb

```diff
@@ -48,9 +44,21 @@ def self.purchase_ticket(conference, quantity, ticket, user)
   def self.update_quantity(conference, quantity, ticket, user)
     purchase = TicketPurchase.where(ticket_id: ticket.id,
                                     conference_id: conference.id,
-                                    user_id: user.id).first
+                                    user_id: user.id,
+                                    paid: 'f').first
 
     purchase.quantity = quantity if quantity > 0
     purchase
   end
+
+  def self.update_paid_ticket_purchases(conference, user, payment)
```

Comment:
This is because the models were not getting access to `current_user` and current `conference` as they are specific to sessions.